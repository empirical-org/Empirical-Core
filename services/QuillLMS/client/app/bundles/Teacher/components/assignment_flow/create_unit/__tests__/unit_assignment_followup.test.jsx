import React from 'react'
import { shallow } from 'enzyme'

import UnitAssignmentFollowup from '../unit_assignment_followup'
import { classroomProps } from '../stage2/__tests__/test_data/test_data'


describe('Unit assignment followup component', () => {

  describe('When there are students assigned', () => {

    describe('on initial load', () => {
      const wrapper = shallow(
        <UnitAssignmentFollowup
          classrooms={classroomProps}
          location={{}}
          referralCode="code"
        />
      )

      it('should render', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('should render the referral form when it loads', () => {
        expect(wrapper.find('.referral').exists()).toBe(true)
      })

    })

    describe('on showNextOptions', () => {
      const wrapper = shallow(
        <UnitAssignmentFollowup
          classrooms={classroomProps}
          location={{}}
          referralCode="code"
        />
      )

      wrapper.setState({ showNextOptions: true, })

      it('should render', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('should render the next options when showNextOptions is true', () => {
        expect(wrapper.find('.next-options').exists()).toBe(true)
      })

    })
  })

  describe('When there are only empty classes assigned', () => {
    const classroomsWithNoStudents = classroomProps.map(c => {
      const newClassroom = c
      newClassroom.students = []
      newClassroom.emptyClassroomSelected = true
      return newClassroom
    })
    const wrapper = shallow(
      <UnitAssignmentFollowup
        classrooms={classroomsWithNoStudents}
        location={{}}
        referralCode="code"
      />
    )

    it('should render', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a referral section', () => {
      expect(wrapper.find('.referral-link-container').exists()).toBe(true)
    })
  })

})
