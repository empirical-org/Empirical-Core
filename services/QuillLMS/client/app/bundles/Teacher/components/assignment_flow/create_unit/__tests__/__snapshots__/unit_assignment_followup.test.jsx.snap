// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit assignment followup component When there are only empty classes assigned should render 1`] = `
<div>
  <AssignmentFlowNavigation
    button={
      <button
        className="quill-button medium contained primary"
        onClick={[Function]}
        type="button"
      >
        Take me to my dashboard
      </button>
    }
  />
  <ScrollToTop />
  <div
    className="container"
  >
    <div
      className="unit-assignment-followup invite-students"
    >
      <h1>
        Success! You assigned 
         to 
        0 classes
        .
      </h1>
      <Card
        header="Invite students to your classes"
        imgAlt="students"
        imgSrc="undefined/images/illustrations/add-students.svg"
        onClick={[Function]}
        text="You currently have 2 classes that have no students."
      />
    </div>
  </div>
</div>
`;

exports[`Unit assignment followup component When there are students assigned on initial load should render 1`] = `
<div>
  <AssignmentFlowNavigation
    button={
      <button
        className="quill-button medium contained primary"
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    }
  />
  <ScrollToTop />
  <div
    className="container"
  >
    <div
      className="unit-assignment-followup referral"
    >
      <Snackbar
        text=""
        visible={false}
      />
      <h1>
        Success! You assigned 
         to 
        1 class
        .
      </h1>
      <div
        className="referral-card"
      >
        <img
          alt="gift"
          src="undefined/images/illustrations/gift.svg"
        />
        <h1>
          Share Quill and earn a month of free Quill Premium for Teachers.
        </h1>
        <p>
          As a nonprofit organization that provides free activities, Quill relies on teachers to share the word.
        </p>
        <p>
          For every teacher that signs up for Quill with your referral link and completes an activity with their students, you earn a month of free Quill Premium, and they receive a one-month free trial.
        </p>
        <div
          className="share-box"
        >
          <div
            className="referral-link-container"
          >
            <Input
              autoComplete="on"
              disabled={true}
              id="referral-link"
              value="undefined/?referral_code=code"
            />
            <CopyToClipboard
              onCopy={[Function]}
              text="undefined/?referral_code=code"
            >
              <button
                className="quill-button secondary outlined small"
                type="button"
              >
                Copy link
              </button>
            </CopyToClipboard>
          </div>
          <p
            className="share-text"
          >
            More ways to share: 
          </p>
          <div
            className="share-links"
          >
            <a
              href="https://twitter.com/home?status=I'm using @quill_org to help my students become better writers and critical thinkers. Want to join me? undefined/?referral_code=code"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="twitter icon"
                src="undefined/images/icons/ui-share-twitter.svg"
              />
            </a>
            <a
              href="https://www.facebook.com/share.php?u=undefined/?referral_code=code"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="facebook icon"
                src="undefined/images/icons/ui-share-facebook.svg"
              />
            </a>
            <a
              href="https://plus.google.com/share?url=undefined/?referral_code=code"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="google plus icon"
                src="undefined/images/icons/ui-share-google.svg"
              />
            </a>
            <a
              href="mailto:mailto:?subject=Free tool to help your students become better writers&body=Hi! I've been using this free tool called Quill.org to help my students become better writers and critical thinkers, and I wanted to let you know about it. Hopefully it helps your students as much as it's helped mine! undefined/?referral_code=code"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="mail"
                src="undefined/images/icons/ui-share-email.svg"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit assignment followup component When there are students assigned on showNextOptions should render 1`] = `
<div>
  <AssignmentFlowNavigation
    button={
      <button
        className="quill-button medium contained primary"
        onClick={[Function]}
        type="button"
      >
        Take me to my dashboard
      </button>
    }
  />
  <ScrollToTop />
  <div
    className="container"
  >
    <div
      className="unit-assignment-followup next-options"
    >
      <h1>
        What would you like to do next?
      </h1>
      <Card
        header="See what I have assigned"
        imgAlt="Clipboard with check"
        imgSrc="undefined/images/illustrations/assigned-activities.svg"
        onClick={[Function]}
        text="View your assigned packs."
      />
      <Card
        header="Invite students"
        imgAlt="Four person icons in a circle"
        imgSrc="undefined/images/illustrations/add-students.svg"
        onClick={[Function]}
        text="Add students to your classes."
      />
      <Card
        header="Assign more activities"
        imgAlt="Stacked pages with plus sign on top one"
        imgSrc="undefined/images/illustrations/assign-activities.svg"
        onClick={[Function]}
        text="Select or build another pack."
      />
      <Card
        header="View as student"
        imgAlt="Magnifying glass around person icon"
        imgSrc="undefined/images/illustrations/view-student.svg"
        onClick={[Function]}
        text="Preview what students will see on their dashboards."
      />
    </div>
  </div>
</div>
`;
